### PR TITLE
Bump to Twig 3

### PR DESCRIPTION
**Pull request by andpavlenko**
While updating to Twig 3 had to remove outdated TwigExtensions. This brought changes in translator which have other format of 'trans' filter. Templates updated to short form `{{ 'Hello'|trans }}`.

Despite the large number of modified files, don't should be critical changes.

### CODE REVIEW

**Comment by gOOvER** [comment]
> Please update composer to no longer include the unused packages Edit: ignore this, I just realized the GitHub app isn't loading a complete list of files that were modified

enable in Preview Features -> Pull Request File Tree

Really usefull Preview Feature

**Comment by BenNottelling** [comment]
> > Please update composer to no longer include the unused packages Edit: ignore this, I just realized the GitHub app isn't loading a complete list of files that were modified
> 
> enable in Preview Features -> Pull Request File Tree
> 
> Really usefull Preview Feature
> 

I have it enabled. I'm just on the mobile app

**Comment by andpavlenko** [comment]
If anybody has no protest, I merge this PR?

File: src/bb-di.php [hunk 1]
```diff
@@ -156,10 +180,12 @@
     // $twig->addExtension(new Twig\Extension\OptimizerExtension());
     $twig->addExtension(new \Twig\Extension\StringLoaderExtension());
     $twig->addExtension(new Twig\Extension\DebugExtension());
-    $twig->addExtension(new Twig\Extensions\I18nExtension());
+    $twig->addExtension(new Symfony\Bridge\Twig\Extension\TranslationExtension());
     $twig->addExtension($box_extensions);
-    $twig->getExtension(Twig\Extension\CoreExtension::class)->setDateFormat($config['locale_date_format']);
-    $twig->getExtension(Twig\Extension\CoreExtension::class)->setTimezone($config['timezone']);
+    $twig->getExtension(Twig\Extension\CoreExtension::class)
```
Comment (BenNottelling) [inline src/bb-di.php:185]: Not sure how other people feel, but I don't like moving this down to the next level like
IMO it hurts the readability of the code

File: src/bb-library/Box/TwigExtensions.php [hunk 2]
```diff
@@ -38,36 +42,43 @@ public function getDi()
...
-
-            'money' => new Twig\TwigFilter('money', 'twig_money', array('needs_environment' => true, 'is_safe' => array('html'))),
-            'money_without_currency' => new Twig\TwigFilter('money_without_currency','twig_money_without_currency', array('needs_environment' => true, 'is_safe' => array('html'))),
-            'money_convert' => new Twig\TwigFilter('money_convert', 'twig_money_convert', array('needs_environment' => true, 'is_safe' => array('html'))),
-            'money_convert_without_currency' => new Twig\TwigFilter('money_convert_without_currency', array('needs_environment' => true, 'is_safe' => array('html'))),
-        );
+        return [
+            /**
+             * 'trans' filter rewrite same filter from Symfony\Bridge\Twig\Extension\TranslationExtension
+             * for compatible with outdated php-gettext library.
```
Comment (BenNottelling) [inline src/bb-library/Box/TwigExtensions.php:48]: Any reason why we didn't just replace it?

**Review by BenNottelling — COMMENTED** [review]
It's hard to review the actually important changes when the majority of the diff here is formatting on unrelated components.
I'd prefer it if large formatting updates like this remained on a different PR or at the least a separate commit by themselves so I can review changes by going between commits

File: src/bb-di.php (hunk 1 above)
Comment (andpavlenko) [reply src/bb-di.php:185]: I think need github precommit hook in each commit! For to keep the code readable.

File: src/bb-library/Box/TwigExtensions.php (hunk 2 above)
Comment (andpavlenko) [reply src/bb-library/Box/TwigExtensions.php:48]: All including in this PR related to templates.
Translation filter relate to localization - its another chain of problems for other PR.

**Pull request merged**